api/lcp/encrypt.py

```diff
@@ -0,0 +1,499 @@
+        """
+        content_id = result_dict.get(cls.CONTENT_ID)
+        content_encryption_key = result_dict.get(cls.CONTENT_ENCRYPTION_KEY)
+        protected_content_location = result_dict.get(cls.PROTECTED_CONTENT_LOCATION)
+        protected_content_length = result_dict.get(cls.PROTECTED_CONTENT_LENGTH)
+        protected_content_sha256 = result_dict.get(cls.PROTECTED_CONTENT_SHA256)
+        protected_content_disposition = result_dict.get(cls.PROTECTED_CONTENT_DISPOSITION)
+        protected_content_type = result_dict.get(cls.PROTECTED_CONTENT_TYPE)
+
+        return cls(
+            content_id=content_id,
+            content_encryption_key=content_encryption_key,
+            protected_content_location=protected_content_location,
+            protected_content_disposition=protected_content_disposition,
+            protected_content_type=protected_content_type,
+            protected_content_length=protected_content_length,
+            protected_content_sha256=protected_content_sha256
+        )
+
+    def __eq__(self, other):
+        """Compares two LCPEncryptorResult objects
+
+        :param other: LCPEncryptorResult object
+        :type other: LCPEncryptionResult
+
+        :return: Boolean value indicating whether two items are equal
+        :rtype: bool
+        """
+        if not isinstance(other, LCPEncryptionResult):
+            return False
+
+        return \
+            self.content_id == other.content_id and \
+            self.content_encryption_key == other.content_encryption_key and \
+            self.protected_content_location == other.protected_content_location and \
+            self.protected_content_length == other.protected_content_length and \
+            self.protected_content_sha256 == other.protected_content_sha256 and \
+            self.protected_content_disposition == other.protected_content_disposition and \
+            self.protected_content_type == other.protected_content_type
+
+    def __repr__(self):
+        """Returns a string representation of a LCPEncryptorResult object
+
+        :return: string representation of a LCPEncryptorResult object
+        :rtype: string
+        """
+        return \
+            '<LCPEncryptor.Result(' \
+            'content_id={0}, ' \
+            'content_encryption_key={1}, ' \
+            'protected_content_location={2}, ' \
+            'protected_content_length={3}, ' \
+            'protected_content_sha256={4}, ' \
+            'protected_content_disposition={5}, ' \
+            'protected_content_type={6})>'.format(
+                self.content_id,
+                self.content_encryption_key,
+                self.protected_content_location,
+                self.protected_content_length,
+                self.protected_content_sha256,
+                self.protected_content_disposition,
+                self.protected_content_type
+            )
+
+
+class LCPEncryptorResultJSONEncoder(JSONEncoder):
+    """Subject's JSON encoder"""
```

Comment:
What does "subject" mean here? We have a data model object called `Subject` but this isn't referring to that.

Reply:
Copy & paste error, updated the docstring